### PR TITLE
Update MiqQE patching

### PR DESCRIPTION
**Pull request by jkrocil**
Purpose or Intent
=================
Update MiqQE patching to be atomic
Add `.miqqe_version` to `IPAppliance`
Add `current_miqqe_version` to `utils.appliance`
`force_navigate` no longer runs MiqQE check on every run, depends on cached `IPAppliance.miqqe_version` instead
Rename `autofocus_upstream` to `autofocus_57`
Fix some lines to reference `self` instead of `store.current_appliance`

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit cdb7f0e599bedcca01d132f46531ab0d6af75c18:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: data/patches/miq_application.js.diff [hunk 1]
```diff
@@ -7,7 +7,7 @@
 +// ---------------
 +
 +// Initialize MiqQE
-+// Make sure to update version here and in pytest_selenium when changed
++// Make sure to update version when changed
```
Comment (jkrocil) [inline data/patches/miq_application.js.diff]: Ironically, when only changing comments, we don't need to do anything, like in this case.

File: utils/appliance/__init__.py [hunk 2]
```diff
@@ -1542,15 +1561,15 @@ def host_id(self, hostname):
 
     def get_yaml_config(self, config_name):
         if config_name == 'vmdb':
-            writeout = store.current_appliance.ssh_client.run_rails_command(
```
Comment (jkrocil) [inline utils/appliance/__init__.py]: This and the instance below might be causing some crazy issues in multi-appliance setups. Potentially in parallelizer as well, not sure if we operate on multiple appliances without switching context anywhere in there...

File: cfme/fixtures/pytest_selenium.py [hunk 3]
```diff
@@ -993,30 +993,13 @@ def force_navigate(page_name, _tries=0, *args, **kwargs):
     # browser fixture should do this, but it's needed for subsequent calls
     ensure_browser_open()
 
-    # check for MiqQE javascript patch in 5.6 on first try and patch the appliance if necessary
-    # raise an exception on subsequent unsuccessful attempts to access the MiqQE javascript funcs
+    # check for MiqQE javascript patch on first try and patch the appliance if necessary
     if store.current_appliance.version >= "5.5.5.0":
-        def _patch_recycle_retry():
+        from utils.appliance import current_miqqe_version
```
Comment (jkrocil) [inline cfme/fixtures/pytest_selenium.py]: Circular import trap avoided!

Comment (psav) [inline cfme/fixtures/pytest_selenium.py]: This will need to be added to navmazing too

File: utils/appliance/__init__.py (hunk 2 above)
Comment (psav) [inline utils/appliance/__init__.py]: It's good to get that stuff changed when we see it

File: cfme/fixtures/pytest_selenium.py (hunk 3 above)
Comment (jkrocil) [inline cfme/fixtures/pytest_selenium.py]: Done.

**Comment by dajoRH** [comment]
Lint report for commit d1f1d579ee3a9fba2661f045b2cd5f117d871072:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 004f58d419722670c6193ba3ab09a17a3b29503e:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by jkrocil** [comment]
Updated and tested with clean 5.7 appliance and #3429.

**Pull request merged**